### PR TITLE
fix: limit number of filtering options for ledger entries.

### LINKED ISSUE #6556: Limit listing of ledger queries to a single Party
Listing ledger entries should be limited to a single party, meaning:
If more than one party is provided for AccountFrom filter - return error
If more than one party is provided for AccountTo filter - return error
If partyID for AccountFrom filter is empty AND partyID for AccountTo is empty -> return error

Consequently queries can include:
- listing ledger entries that one party made to some * parties
- listing ledger entries that one party received from some * parties
- listing ledger between two parties

### PR DESCRIPTION
**Pull request by AnExsomnis**
close: #6556

### CODE REVIEW

**Review by peterbarrow — DISMISSED** [review]
All good once you remove the binary file `vega` from your checkin

**Comment by AnExsomnis** [comment]
> All good once you remove the binary file `vega` from your checkin

Removed, thanks! sorry

**Review by peterbarrow — DISMISSED** [review]

File: datanode/sqlstore/ledgerentry_filter.go [hunk 1]
```diff
@@ -180,3 +137,28 @@ func prepareGroupFields(groupByAccountField []entities.AccountField, groupByLedg
 
 	return fields
 }
+
+var ErrLedgerEntryFilterForParty = fmt.Errorf("filtering ledger entries should be limited to a single party")
```
Comment (jeremyletang) [inline datanode/sqlstore/ledgerentry_filter.go:141]: We usually add those at the tops of files. also seing there's not formatting directiver, errors.New() should suffice.

Comment (AnExsomnis) [reply datanode/sqlstore/ledgerentry_filter.go]: Addressed here https://github.com/vegaprotocol/vega/pull/6557/commits/92438786697c4248d2d0c9c989d79eaf938c7778#diff-149a70d32b3e4f50b3d653208267b371f6335459774f97eb3dd122023e3432fcR11-R12
Thanks!

**Review by jeremyletang — APPROVED** [review]

**Pull request merged**